rules/gsutil.bzl

```diff
@@ -0,0 +1,71 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Skylark rule for pulling a file from GCS bucket.
+"""
+def _impl(repository_ctx):
+  """Core implementation of gsutil_cp."""
+  target_file = repository_ctx.attr.target_file or repository_ctx.attr.file
```

Comment:
you need to validate that the target_file string passed is a valid path (e.g., see https://github.com/bazelbuild/bazel/pull/5647/files)

Reply:
Will follow your PR and make such validations first.